### PR TITLE
added newline for better ANSI output

### LINKED ISSUE #898: Better Ansi Output
## Description
We've just started using phpmd in an already established codebase and the ANSI output was just confusing to read or to understand what belongs to what.

It's unclear for users who took the first look at the output that it's divided like
```
File:
-----
Findings
```

But instead of being read as
```
-----
Findings
File:
-----
```

Adding a blank like certainly would make the output more readable and easier on the eyes, like this:
| Before | After |
|--------|-------|
|Screenshot 2021-06-10 at 11 36 14 AM |Screenshot 2021-06-10 at 11 43 26 AM|

I have not looked into the codebase to say if I could provide a PR.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] Tell if you have the option to provide the code for this proposal.
* [x] This issue is about 1 feature proposal and nothing more.
* [x] The issue has a descriptive title. For example: "Add JSON render option".

### PR DESCRIPTION
**Pull request by himanshu007-creator**
Type: (bugfix / feature / refactoring / documentation update)  
Issue: Resolves #898 

Added a newline to make output better

### CODE REVIEW

**Review by tvbeek — COMMENTED** [review]
Thanks for the PR, it is a good start. Founding where you need to be is a big part of the work. I have a little comment on the code to make it more in line with the other parts.

Do you have the option to add a test?

File: src/main/php/PHPMD/Renderer/AnsiRenderer.php [hunk 1]
```diff
@@ -38,6 +38,7 @@ private function writeViolationsReport(Report $report)
         $previousFile = null;
         foreach ($report->getRuleViolations() as $violation) {
             if ($violation->getFileName() !== $previousFile) {
+                echo nl2br("\n");
```
[the remark this answers is not in the document]
Comment (kylekatarnls) [reply src/main/php/PHPMD/Renderer/AnsiRenderer.php:41]: Why `nl2br()`? ANSI output should not contain HTML.

[the remark this answers is not in the document]
Comment (himanshu007-creator) [reply src/main/php/PHPMD/Renderer/AnsiRenderer.php]: @tvbeek done

File: src/main/php/PHPMD/Renderer/AnsiRenderer.php [hunk 2]
```diff
@@ -38,6 +38,7 @@ private function writeViolationsReport(Report $report)
         $previousFile = null;
         foreach ($report->getRuleViolations() as $violation) {
             if ($violation->getFileName() !== $previousFile) {
+                $this->writer->write('\n');
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Renderer/AnsiRenderer.php:41]: We don't want an extra empty line for the first one:
```suggestion
                if ($previousFile !== null) {
                    $this->writer->write('\n');
                }
```

**Review by kylekatarnls — APPROVED** [review]

File: src/main/php/PHPMD/Renderer/AnsiRenderer.php [hunk 3]
```diff
@@ -38,6 +38,9 @@ private function writeViolationsReport(Report $report)
         $previousFile = null;
         foreach ($report->getRuleViolations() as $violation) {
             if ($violation->getFileName() !== $previousFile) {
+                if ($previousFile !== null) {
+                    $this->writer->write('\n');
+                }
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Renderer/AnsiRenderer.php:43]:
```suggestion
                    $this->writer->write(PHP_EOL);
                }

```

**Review by kylekatarnls — APPROVED** [review]

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**